docs/shell-integration.md

```diff
@@ -180,3 +181,33 @@ return config
 
 Now, rather than just running `cmd.exe` on its own, this will cause `cmd.exe`
 to self-inject the clink line editor.
+
+## OSC 52 Clipboard Copy
+
+The data can be copied to the `System Clipboard` or `Primary Selection` with a
+command like this:
+
+```bash
+printf '\033]52;c;%s\033\\' $(base64 <<< "hello world")
+```
+
+- `c` copies to the system clipboard
+- `p` copies to the primary selection buffer
+
+The second parameter provides the selection data, which is a string encoded in base64 (RFC-4648).
+
+## OSC 52 Clipboard Paste
+
+The data can be pasted from the `System Clipboard` or `Primary Selection` with a
+command like this:
+
+```bash
+printf "\033]7;c;?\033\\"
```

Comment:
I'm trying to test this out and the above didn't do anything. Surely it's supposed to be:
```suggestion
printf "\033]52;c;?\033\\"
```
which results in wezterm sending the base64ed clipboard to the screen. How would I capture/de-base64 it?

Reply:
BTW, my goal is to write a bash function that will output the clipboard so I can bind that to a key to insert via readline. Essentially I want something like this:
```bash
function _shell_paste_p() {
    printf "\033]52;p;?\033\\"
}
_rl_paste_p() {
    CLIP=$(_shell_paste_p)
    COUNT=$(echo -n "$CLIP" | wc -c)
    READLINE_LINE="${READLINE_LINE:0:$READLINE_POINT}${CLIP}${READLINE_LINE:$READLINE_POINT}"
    READLINE_POINT=$(($READLINE_POINT + $COUNT))
}
bind -m emacs -x '"\C-y": _rl_paste_p'
```
The "cut line" equivalent is this (which works fine):
```bash
function _shell_copy() {
    printf "\033]52;c;$(base64 | tr -d '\r\n')\a"
}
_rl_kill() {
    echo -n "${READLINE_LINE:$READLINE_POINT}" | _shell_copy
    READLINE_LINE="${READLINE_LINE:0:$READLINE_POINT}"
}
bind -m emacs -x '"\C-k": _rl_kill'
```

Reply:
Thanks for fixing the typos! Typically, decoding is handled by editors. For example, in Neovim https://github.com/neovim/neovim/blob/2024c10d74cea760ff9e4ac8208cc4c9c176f4d3/runtime/lua/vim/ui/clipboard/osc52.lua#L22. There's also a command-line utility for this: https://github.com/theimpostor/osc. You need to read the tty device directly to obtain the response https://github.com/theimpostor/osc/blob/8a313c60d67353669509e9461d22b403df72b5e7/main.go#L273. I’m not sure how to do this in Bash myself, so if you manage to decode the response successfully, please share your solution!

Reply:
Thanks for pointing me at the command line `osc` tool. I've got my bash integration working now with:
```
function _shell_paste_c() {
    osc paste
}
_rl_paste_c() {
    CLIP=$(_shell_paste_c)
    COUNT=$(echo -n "$CLIP" | wc -c)
    READLINE_LINE="${READLINE_LINE:0:$READLINE_POINT}${CLIP}${READLINE_LINE:$READLINE_POINT}"
    READLINE_POINT=$(($READLINE_POINT + $COUNT))
}
bind -m emacs -x '"\C-y": _rl_paste_c'
```
(I gave my earlier functions a "_p" suffix as I was expecting to use the primary selection -- the osc tool doesn't seem to let you choose primary vs clipboard, so I'm using the clipboard instead (hence I named the functions with "_c" suffix). It's all working fine, thanks!